nilearn/decoding/decoder.py

```diff
@@ -0,0 +1,849 @@
+
+    def _binarize_y(self, y):
+        """Helper function invoked just before fitting a classifier."""
+        y = np.array(y)
+
+        # encode target classes as -1 and 1
+        self._enc = LabelBinarizer(pos_label=1, neg_label=-1)
+        y = self._enc.fit_transform(y)
+        self.classes_ = self._enc.classes_
+        self.n_classes_ = len(self.classes_)
+        return y
+
+
+class DecoderRegressor(BaseDecoder):
+    """A wrapper for popular regression strategies in neuroimaging.
+
+    The `DecoderRegressor` object supports regression methods.
+    It implements a model selection scheme that averages the best models
+    within a cross validation loop. The resulting average model is the
+    one used as a regressor. The `Decoder` object also leverages the
+    `NiftiMaskers` to provide a direct interface with the Nifti files on disk.
+
+    Parameters
+    -----------
+    estimator : str, optional
+        The estimator to choose among: 'ridge', 'ridge_regression', and 'svr'.
+        Note that the 'ridge' and 'ridge_regression' correspond to the same
+        estimator. Default 'svr'.
+
+    mask : filename, Nifti1Image, NiftiMasker, or MultiNiftiMasker, optional
+        Mask to be used on data. If an instance of masker is passed,
+        then its mask and parameters will be used. If no mask is given, mask
+        will be computed automatically from provided images by an inbuilt
+        masker with default parameters. Refer to NiftiMasker or
+        MultiNiftiMasker to check for default parameters. Default None
+
+    cv : cross-validation generator or int, optional (default 10)
+        A cross-validation generator. If None, a 3-fold cross
+        validation is used for regression.
+
+    param_grid : dict of str to sequence, or sequence of such. Default None
+        The parameter grid to explore, as a dictionary mapping estimator
+        parameters to sequences of allowed values.
+
+        An empty dict signifies default parameters.
+
+        A sequence of dicts signifies a sequence of grids to search, and is
+        useful to avoid exploring parameter combinations that make no sense
+        or have no effect. See scikit-learn documentation for more information.
+
+    screening_percentile : int, float, optional, in the closed interval [0, 100]
+        Perform an univariate feature selection based on the Anova F-value for
+        the input data. A float according to a percentile of the highest
+        scores. Default: 20.
+
+    scoring : str, callable or None, optional. Default: 'r2'
+        The scoring strategy to use. See the scikit-learn documentation
+        If callable, takes as arguments the fitted estimator, the
+        test data (X_test) and the test target (y_test) if y is
+        not None.
+        e.g. scorer(estimator, X_test, y_test)
+
+        For regression: 'r2', 'mean_absolute_error', or 'mean_squared_error'.
+            Default 'r2'.
```

Comment:
```suggestion
        Default 'r2'.
```
I am not sure why this line is indented.